action/protocol/poll/nativestaking.go

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2019 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package poll
+
+import (
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/iotexproject/iotex-address/address"
+	"github.com/iotexproject/iotex-election/types"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/action/protocol"
+)
+
+var dummyCaller, _ = address.FromString(address.ZeroAddress)
+
+type (
+	// NativeStaking represents native staking struct
+	NativeStaking struct {
+		cm       protocol.ChainManager
+		contract string
+		abi      abi.ABI
+	}
+
+	pygg struct {
+		Count           *big.Int
+		Indexes         []*big.Int
+		StakeStartTimes []*big.Int
+		StakeDurations  []*big.Int
+		Decays          []bool
+		StakedAmounts   []*big.Int
+		CanNames        [][12]byte
+		Owners          []common.Address
+	}
+)
+
+// NewNativeStaking creates a NativeStaking instance
+func NewNativeStaking(cm protocol.ChainManager, staking string) (*NativeStaking, error) {
+	abi, err := abi.JSON(strings.NewReader(NsAbi))
+	if err != nil {
+		return nil, err
+	}
+	return &NativeStaking{cm, staking, abi}, nil
+}
+
+// Votes returns the votes on height
+func (ns *NativeStaking) Votes(height uint64) ([]*types.Vote, error) {
+	if ns.contract == "" {
+		return nil, ErrNoData
+	}
+
+	// read voter list from staking contract
+	votes := []*types.Vote{}
+	prevIndex := big.NewInt(0)
+	limit := big.NewInt(256)
```

Comment:
check if it's uint8, if it is, 256 will overflow

Reply:
it is uint256